components/Base/CollapsableItem.vue

```diff
@@ -0,0 +1,37 @@
+<template>
+  <Disclosure v-slot="{ open }">
+    <div
+      class="flex justify-between h-14 w-full border-b border-solid border-b-gray-200"
+    >
```

Comment:
What I mean is according to docs what you implemented here would be like:
```html
<template>
  <div>
     <button>
       text with icon
     </button>
  </div>
  <div>
    description
  </div>
</template>
```
So top level is not a button if this is what you intended to do.

And using `as="div"` we could get:
```html
<div>
  <div>
     <button>
       text with icon
     </button>
  </div>
  <div>
    description
  </div>
</div>
```
And maybe thanks to that we could remove one of those divs 🤔 
I'll run this locally tomorrow and check if I'm correct because i think this could be even further simplified thanks to Headles UI API.

Reply:
I take you point. I this case this structure worked out ok since we making them into list